protocol-units/bridge/chains/ethereum/src/client.rs

```diff
@@ -223,18 +250,44 @@ impl BridgeContractInitiator for EthClient {
 		time_lock: TimeLock,
 		amount: Amount, // the ETH amount
 	) -> BridgeContractInitiatorResult<()> {
+		if amount.weth() > 0 {
+			let deposit_weth_signer = self.get_signer_address();
+			println!("deposit_weth_signer: {:?}", deposit_weth_signer);
+			let contract = self.weth_contract().expect("WETH contract not set");
+			let call = contract.deposit().value(U256::from(amount.weth()));
+			send_transaction(call, &send_transaction_rules(), RETRIES, GAS_LIMIT)
+				.await
+				.expect("Failed to deposit eth to weth contract");
+
+			let approve_call: alloy::contract::CallBuilder<_, &_, _> = contract
+				.approve(self.initiator_contract_address()?, U256::from(amount.weth()));
+			let WETH9::balanceOfReturn { _0: balance } = contract
+				.balanceOf(deposit_weth_signer)
+				.call()
+				.await
+				.expect("Failed to get balance");
+			let signer = self.get_signer_address();
+			println!("caller: {}", signer);
+			println!("balance: {}", balance);
+
+			send_transaction(approve_call, &send_transaction_rules(), RETRIES, GAS_LIMIT)
+				.await
+				.expect("Failed to approve");
+		}
```

Comment:
All this logic inside `if amount.weth() > 0 ` I would put inside the integration test. It's not the responsibility of this function to do WETH calls, its just for calling the bridge contracts. 

In real world if a user hasn't called approve on a WETH contract we shouldn't be handling that for them off-chain in a hidden way. 

Put all this logic inside the `eth_movement.rs` integration test, perhaps in a utility function so that you can reuse it. It could even be a method on the `TestHarness`, as its technically a testing utility method.